### PR TITLE
Changed implant functionality

### PR DESCRIPTION
**Pull request by MacHac**
Refer to issue #733.
### Intent of Pull Request

This pull request modifies loyalty implanter functionality.  At the moment, traitors and changelings can be loyalty implanted without problems.  This makes security recruitment incredibly dangerous when another type of antagonist hasn't been confirmed.  With this PR, all special roles will now properly reject loyalty implants.  To help traitors counter this, the freedom implant (which now costs 7 crystals instead of 5) will quietly deactivate loyalty implants as they are injected, similar to trying to implant cybermen.

### CODE REVIEW

**Comment by ToGWtF** [comment]
Don't make changelings resist that's retarded

**Comment by MacHac** [comment]
Not really.  It makes sense from both a game balance and a lore standpoint.  Sure, changelings are very different, but if you read the lore that nobody reads it shows that they were bio-engineered by the syndicate from humans.  Plus, the loyalty implant would probably signal that something was up even if changelings were fundamentally different enough to be incompatible with an implant.  And from a game balance standpoint, changelings are easy as hell right now even without the ability to waltz into security sporting their lovely new implant.

**Comment by ghost** [comment]
Don't make changelings resist it. 

A core part of changelings is that they can't replicate metal-Right from the thing. You can't transform to avoid implants.

So no.

Don't make changelings resist loyal implants, PLEASE.

Just PLEASE.

Traitors resisting is fine. PLEASE not changelings. PLEASE.

**Comment by Cruix** [comment]
Adam, what about traitor lings?

**Comment by ShadowDeath6** [comment]
> traitorling
> still ling
> but traitor
> ebin

**Comment by FluffySurvivor** [comment]
This is stupid and a huge buff to security. 
Security will now just use implants to check people for antag. 
:-1: x :100:

**Comment by ParagonPrime** [comment]
no, Fluffy, because there aren't enough implants for that

what this does is make it a lot harder for stupid HoPs who promote anyone who asks for sec access to fuck up everything for sec

**Comment by X-TheDark** [comment]
Plus, this deals with the exceptional "super speshul snowflek" unwritten meta-shield regarding how loyalty implants interact with traitorship.

**Comment by DavidEdwards** [comment]
Needs more input - more balance discussion.

**Comment by ShadowDeath6** [comment]
I'd say keep the cost of freedom implant at 5, this isn't so big of a buff that it makes the item any more viable than it already is. 

### Please make lings just take the implant, their point is to be sneaky breeky.

**Comment by ParagonPrime** [comment]
let lings take the implant, BUT it has a chance to degrade whenever they use ling abilities (any of them)

**Comment by ToGWtF** [comment]
Getting a loyalty implant as a ling is basically forcing you to stay as your current identity. You change forms when caught? You still have the implant and they know its you right away. Getting a loyalty implant as a ling is already such a pain in the ass don't make them get outed and executed for resisting

**Comment by EMT321** [comment]
At least make traitors resist the implants, its tiring having a guy get implanted then shoot you to death 30 seconds later.

**Comment by ghost** [comment]
sec will just

implant everyone they arrest for any reason

breaking into the HOP office? implant

reported for any crime? implant.

security becomes mass tazeimplant turbohitlers implanting everyone they see and we have to add a rule for them to not do that.

makes the detective useless, security won't need him if they can buy an automatic 'this is definitely absolutely not an antag' button. any crime at all is an implant.

also if the freedom implant makes the thing silent they'll still notice the missing N and kill you anyway.

I can't stress enough how shit that is. Sec will, and they WILL-this isn't a hypothetical scenario-implant almost anyone they arrest at ALL. 

ANYONE.

Frame someone to hopefully get them arrested by sec? too bad, sec implanted them, they know they DEFINITELY didn't do it, go kill you

any minor thing you do is liable to get you forceimplanted, resisted, and executed on the spot.

it's fucking bad, stop this right now.

stop it.

if you don't want to risk someone being an antag, don't hire them, or better yet put chem implants in them so if they ARE a traitor they die a screaming death shortly later.

or hire someone you can actually trust.

the intent behind this is to stop traitor sec. but what it will end up with is sec mass implanting anyone they arrest at all, even for tresspassing-at the least they'll do this early round, and so if you get arrested for busting into tool storage for a multitool while a traitor and haven't bought anything, you still get lolimplant resisted executed cremated.

**Comment by ParagonPrime** [comment]
then let's modify rules/space law on implants

you can only implant under two circumstances:
1. recruiting someone for sec/head position
2. confirmed subversion of chain of command (so rev/gangs)

that'll stop wanton implanting dead in its tracks
case closed

**Comment by EMT321** [comment]
In response to Adam: Things ain't cheap. You start with 8. Force-implanting the crew is difficult enough when that is all cargo is doing, it is an exaggeration to suggest that it is a viable tactic for traitorling rounds. Sec could use a buff in this regard: I've strolled right into armory as an implanted traitor, because they had no way of knowing I wasn't on the level.

**Comment by ShadowDeath6** [comment]
> space law ever being followed by sec when they deal with traitors
> buttmad traitors crying about space law not being followed

hahahahahahahaha

**Comment by ghost** [comment]
but sec will just order implants from cargo every round

or implant anyone they brig early game/implant all heads of staff at roundstart just to ensure none of them are traitors

i can guarantee you it'll end up with all heads of staff being force implanted at roundstart and outing them as traitors

> they had no way of knowing I wasn't on the level

THIS IS THE POINT OF TRAITOR.

Security NEEDS NO MORE BUFFS. They are already powerful.

If you don't want implanted traitors killing sec with the armory don't make non-trustworthy people sec.

**Comment by ghost** [comment]
but sec will just order implants from cargo every round

or implant anyone they brig early game/implant all heads of staff at roundstart just to ensure none of them are traitors

i can guarantee you it'll end up with all heads of staff being force implanted at roundstart and outing them as traitors

> they had no way of knowing I wasn't on the level

THIS IS THE POINT OF TRAITOR.

Security NEEDS NO MORE BUFFS. They are already powerful.

If you don't want implanted traitors killing sec with the armory don't make non-trustworthy people sec.

**Comment by MacHac** [comment]
Implant crates currently cost 40 supply points.  This can be increased if necessary.  There's already enough greytiding going on that it's not reasonable to expect security to be able to implant every criminal in a round.

**FOR TRAITORS:**
Now there's more reason to drop the points for a freedom implant.  Those who don't will get caught by the loyalty implants, _**WHICH MAKES SENSE.**_

**FOR CHANGELINGS:**
It's a nerf.  They need one.

**Comment by TheSadFox** [comment]
Or simply stop trying to shove an unneeded feature that will only cause massive balance issues down people throats.

Traitors being hired as sec has hardly ever been a problem anyways.

**Comment by EMT321** [comment]
Adam, you are overestimating the issue significantly. The cost of implanting everyone ever is too high for a non-dedicated force managing cargo, and requires cargo's willingness as well. the chances of it happening on a typical traitor round are little to none.

As far as issues go, I've personally seen about a dozen instances of traitors joining sec. I do it myself frequently.

As far as your contention "Dont let non-trustworthy people play sec", how do you suppose one gets more officers when the HoS/Warden starts alone?

**Comment by ParagonPrime** [comment]
You want fucking tatorsec? Work for it. Take a bit of a risk like EVERY OTHER TATOR JOB ON STATION SAVE AI AND HoP WITHOUT A CAPTAIN.

Tatorsec is currently many TC's worth of gear, gotten for absolute free on a slight gamble by asking the HoP. That ain't right, not one bit. 

Hell, if loyalty implants don't even work on syndicate agents, why don't we start the shift with traitor sec?

Oh, wait, I know why. It's unbalanced as shit.

**Comment by MacHac** [comment]
Quite frankly, I'm not sure why this is so controversial.  It makes sense from a game balance standpoint, and from a logical standpoint too.  Why would NT drop mad supply points on keeping security implanted if the implants didn't even work?  I honestly believe that this was an oversight.

> SECURITY WILL JUST IMPLANT EVERYONE
> The fact that implanting everyone could actually be a viable strategy aside, security can't implant everyone.  This is because, at peak hours, security could barely hit 10% of the station with their roundstart implants.  Getting more requires talking cargo into spending most of their supply points over the round on getting more implants, and even then implant flow would be too slow and people too slippery to make the strategy effective.  So what if secling doesn't work any more?  Lings are unbalanced, and lings with sec privileges even more so.  As for traitors, just drop the TC on a freedom implant.

**Comment by ghost** [comment]
Security can implant everyone, play some rev.

Sec barges into cargo, stuns the techs, implants them, and order the miners to get shitloads of plasma, and suddenly from the upgraded smelter you get hundreds of points and get 6 loyalty implants in minutes. Or, alternately, 3 loyalty implants every 4 minutes, where you are INSTANTLY assured they are not a traitor if implanted.

Anyone sec brigs is gonna get implanted at roundstart, and heads of staff are ALWAYS going to get implanted, especially the HoP, every round, at roundstart.

And then sec will barge into cargo and order implants constantly. This happens during rev, with 30-40% of the station implanted in a chaotic roundtype with officers getting killed off. It'll happen during traitor, except now you have storms of people running around acting as sec officers versus one or two terrified traitors.

'drop the TC on a freedom implant'

so make freedom implants required gear at roundstart, then? Because otherwise if you don't you're gonna get fucked the moment you get implanted, so it's about as much required gear as an emag. You're just forcing people to buy something because of a shitty nerf you made that's unneeded.

Traitors have been fine for three years, there's no need to nerf them now.

This is quite frankly not happening. It's not going to be 'secling doesn't work anymore.' You're denying people a viable strategy, and you're buffing security to a ridiculous extent because yes, they WILL force implant everyone, or at least spam implant a lot of people by breaking into cargo, ordering implants, and mass implanting everyone important they see or anyone they recognize as robust, including heads and such. Constantly. Every fucking department.

Imagine what a nightmare this would be on lowpop, with 30 people! HoS grabs implants from armory and his locker, runs into the hall, stuns all of medical one by one, force implant force implant force implant everyone, and he's assured that A. Either medical is completely not traitors, or B. He finds a traitor and kills him, no counter.

It forces you to spend 5 TC on a freedom implant at the start of the round, every round, because if you're so much has hacking into aux tool storage within the first 10-20 minutes, you'll get stunned and force implanted, ESPECIALLY if security isn't in need of any officers.

You're low on staff? Hire some more and take the risk, hire people who are trustworthy and take less risk, or don't hire any. We don't have automatic non-antag engineer dispensers if there are no engineers, security doesn't need a stupid buff like this.

Also if you implant a revhead at roundstart and he gets executed sec can't call rev because it could be traitors, which is a metashield revs don't need, OR if they CAN call rev then they start spamimplanting and completely avert the rule Paragon mentioned adding above because 'lol how did I know it's not traitor!?!'

DO you understand the balance connotations of this? Have you ever played a game in your LIFE?

Because if you're arguing this, you really have not. Implants are very easy to order from cargo, they start with six in the brig, and you have all the time in the world to implant people during traitor, as compared to rev. 

During rev, you're pressed for time, heads are being attacked, ect. It takes time to implant people. During traitor, you have all the time in the world to take over cargo, hijack it for implant crates, and spam order, all under the justificaiton of 'some guy resisted i thot it was rev'

And if you add a rule like 'don't implant w/o need' then that leads to people getting banned because they implanted a traitor because they legitimately thought it was rev due to getting disarmed a bunch by some assistants and all that, which is more shit we really don't need.

No. This is a dumb change, to something that does not, and has never, needed fixing. 

You don't want traitor sec, don't be an idiot with who you hire, or DON'T HIRE NEW SEC.

Christ.

You're more sec buff than Steel 'I wanna make security judge dredd' point over on /tg/. This is prepostrous and quite frankly i'm not letting it get merged no matter what you do.

It's not happening.

You're forcing people to waste TC on an implant at roundstart or risk getting fucked instantly by sec because you got arrested breaking into aux tool storage or got spam caught and implanted by sec from cargo.

You're fucking over a changeling who manages to kill and absorb the officer, because even if they go to all the trouble of fabricating a story that they got killed and cloned and get welcomed back to security, you still get implanted, killed, and cremated on the spot, especially because lings don't have that.

You're not 'talking' cargo into spending points. YOu're muscling in, force implanting cargo, and hijacking the console every round to spend points on implants, every round, and implant anyone you arrest or find mildly suspicious, and if mining brings back plasma, or you get data disks from research, or botany is growing anything, you ship off the goods-and the crates-for points and get shitloads more points to order implant crates, ESPECIALLY if the smelter is upgraded and you get 400 sheets of plasma in 3 minutes, which is literally how it works.

And if you add a rule to restrict implanting that leads to confusion, people getting banned for implanting traitors they found hacking into perma because 'its without reason', banned for implanting a ling because they thought it was rev due to getting disarmed a bunch, ect ect, and thus terrified to actually implant people when it IS gang or rev, and dying horribly in the fucking process.

And if you make implants cost more from cargo, then you're fucking buffing revs and gangs into becoming even MORE untouchable by security than they already are!

This is ridiculous and badly thought out. Honestly one of the worst ideas i've ever heard since that stupid antag mice shit. The amount of stuff wrong with it I have gone into above.

It's not an issue of traitor security-And traitor security isn't a fucking issue, since everyone knows when a sec officer goes rogue and a borg beats the shit out of them, or  gasp, they're actually robust and fooled their way into being a sec officer and acting trustworthy enough to get promoted to one, and then managed to security, better ban them.

Either don't be such a blind idiot when hiring security and keep an eye on them, or don't hire new officers. No need to nerf traitors and lings with a random lol gg insta counter gg meme.

**Comment by EMT321** [comment]
"DO you understand the balance connotations of this? Have you ever played a game in your LIFE?"

Yes, I have. And so have the other plethora of people who feel this is a useful addition. Being talked down to and having concerns marginalized because, in your eyes, "traitor security isn't a fucking issue" does not resolve my (and i'm lead to believe shared with others) concerns, simply that you can gain sec access and gear at no real risk to yourself as a traitor. It's something I've witnessed, something I've done to great effect myself, and something which seems at odds with traitors having to strike a balance between risk and resources available.

As far as the idea that the new meta is muscling into cargo and force-implanting everyone while commandeering control, not only will that provoke greytide  and legitimate claims of shitcurity against you, but it also practically necessitates detailing at least one officer to suppress cargo from ordering anything else for the station. That's one less officer out of a likely understaffed team to respond to literally anywhere else, and a team of cargo techs/QM/Miners who are sure to be less than thrilled by the goon who has to waste time holding a gun to their head.

"You don't want traitor sec, don't be an idiot with who you hire, or DON'T HIRE NEW SEC"

Unless you want sec to metafriend hardcore on who gets promoted as being "trustworthy", or have more rounds where you have a total of 3 security staff and no warden for 60+ people, there needs to be some manner by which I can prevent being gunned down by my own people, while possessing a manner in which to add to their ranks. 

Paragon put it well: We dont have traitor sec/detectives for a reason. At the moment there is literally no defense from the guy who signs up for a legitimate job change, gets implanted, then stuns and kills the HoS/Warden from behind as they're standing in brig and robs the armory blind.

For the record, I dont agree with changelings resisting implants.

**Comment by ghost** [comment]
Except for the 99 officers who are prepared to immediately kill newhire sec.

Also you still didn't adress my points about complicating rev/gang and resisting implants there and the issues the new rules would cause.

Oh and also you know, since you're implanted you can't rebel, if you force implant the cargo techs and you tell them to order implants, they HAVE to order implants or they get banned, basically. Cargo can't say 'nah fuck you' and ignore you, they have to listen to you when implanted, meaning that you don't need an officer in there, you just force implant them, and force implant anyone greytiding, and force implant all the heads, and suddenly there's no greytide or anything because it turns into 'obey or get banned. and the people fighting against sec get implanted and forced to obey them.

Basically? Get good. If you really don't trust your own officers chem implant them.

**Comment by EMT321** [comment]
"Oh and also you know, since you're implanted you can't rebel"

There's a big difference between ordering what you/the station wants, and committing an insurrection. I see implanted people greytide on gangs frequently.

And rather than dismissing my points with "get good", can you provide another example of a traitor job where they get access to stunlocks, flashbang protection, and full maint access for free? The armory is one c4 away if you want to be stealthy, and it's easy enough to take and kill the warden as he's adjusting a prisoners sentence.

Threatening officers with death if they mess up just leads to the clown and greytiders accusing them of things, and people less likely to play sec. There should be a middle ground between death threats to your own team, and implants which do nothing versus traitors.

**Comment by ghost** [comment]
Ahelp if they greytide on gang because they're not supposed to and ban if they do.

That's like arguing that i've seen toxins grief so remove toxins entirely

also, scientist has all of the above and stronger weaponry

botany has all of the above and stronger weaponry

miner is completely unkillable in space

engineer has access wherever the fuck he wants and flashbang proof hardsuits 

atmos tech literally warms his bum in fire

sec is powerful, yeah, but it takes one poke of a stupid officer to get his shit as any other job and there are more than a few jobs that are stupidly strong enough anyway.

cargo tech can get sec gear in minutes, all the shit they normally can, and doesn't have to deal with the HoS watching his every step, and science is...
well...

science.

**Comment by ghost** [comment]
shit, how did i forget

**Comment by ShadowDeath6** [comment]
Wonder what the wordcount is on this issue. Honestly after thinking this over it's basically making traitors more likely to do stupid murderboner bullshit than actually get into sec and do stuff from there, which isn't very interesting. I admit, loyalty implants go a very long way. I managed to dodge sec all round as a traitor simply with an agent ID, voice changer, and an implant. 

> lings need a nerf

Honestly, stealth lings don't need much of a nerf, it's their murdery abilities that make them super powerful. This would be a massive nerf to lings who actually try to work their way into sec since they get implanted and it's laser to the face + cremate / gibber / spaced while cuffed. 

> remove toxins entirely

Good idea adam

**Comment by FluffySurvivor** [comment]
or how about you roll HoS and kick the shit out of the random geneticist that's in your equipment room while he is screaming "BBBUT HOP MADE ME SEEEEC" and leave him naked and cuffed in front of brig. After you're done with the guy, you proceed with the hop. Literally covered in space law. 
/nice/ security doesn't get you anywhere 

Personally, as HoP, I never recruit security unless the antags are confirmed. It's always fun to say NO when the assistants ask for security and then they go like "b-but wai, i want sec ;((" 

anyway, this suggestion is ultra dumb. Antagsec can be countered. 

Adam covered it nicely

**Comment by MacHac** [comment]
In hindsight, I think this was unnecessary.  Closing for now.
